rust/experimental/query_engine/expressions/src/scalars/text_scalar_expression.rs

```diff
@@ -8,6 +8,12 @@ pub enum TextScalarExpression {
     /// Returns a string with all occurrences of a lookup value replaced with a
     /// replacement value or null for invalid input.
     Replace(ReplaceTextScalarExpression),
+
+    /// Returns true if regex expression matches the input.
+    Match(MatchTextScalarExpression),
+
+    /// Returns the first string value for a named capture in a regex for the input string
```

Comment:
Should also mention the index bit. Something like...

```suggestion
/// Returns the first string value for a named capture or a capture index using the provided regex and input string.
```